test/rekt/resources/jobsink/jobsink.go

```diff
@@ -106,6 +106,16 @@ func WithJob(job batchv1.Job) manifest.CfgFn {
 	}
 }
 
+func WithPodIstioAnnotations() func(*batchv1.Job) {
+	return func(job *batchv1.Job) {
+		job.Spec.Template.Annotations = map[string]string{
+			"sidecar.istio.io/inject":                "true",
```

Comment:
while the annotation works, it's deprecated and they recommend a label: https://istio.io/latest/docs/setup/additional-setup/sidecar-injection/#controlling-the-injection-policy

To avoid having to change it when they drop it or being incompatible with older versions, we were using both